### PR TITLE
MAKE-827: cli for downloading files

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: cli/command.go [hunk 1]
```diff
@@ -368,3 +368,64 @@ func KillAllCMD() cli.Command {
...
+		pathFlagName        = "path"
+		extractPathFlagName = "extract_to"
+	)
+
+	return cli.Command{
+		Name: "download",
+		Flags: append(clientFlags(),
+			cli.StringFlag{
+				Name:  joinFlagNames(urlFlagName, "p"),
+				Usage: "specify the path (URL) of the file to download on the remote.",
```
Comment (Kimchelly) [inline cli/command.go]: Can we just say URL since the current description sounds like a filepath?

File: cli/command.go [hunk 2]
```diff
@@ -368,3 +368,64 @@ func KillAllCMD() cli.Command {
...
+				Name:  pathFlagName,
+				Usage: "specify the remote path to download the file to on the managed system. Required.",
+			}),
+		Before: mergeBeforeFuncs(
+			clientBefore(),
+			requireStringFlag(pathFlagName),
+			func(c *cli.Context) error {
+				if c.String(urlFlagName) == "" {
+					if c.NArg() != 1 {
+						return errors.New("must specify a path")
```
Comment (Kimchelly) [inline cli/command.go]: "specify a URL"

File: cli/command.go [hunk 3]
```diff
@@ -368,3 +368,64 @@ func KillAllCMD() cli.Command {
...
+					}
+					return errors.Wrap(c.Set(urlFlagName, c.Args().First()), "problem setting path from positional flags")
```
Comment (Kimchelly) [inline cli/command.go]: path -> URL

File: cli/command.go [hunk 4]
```diff
@@ -368,3 +368,64 @@ func KillAllCMD() cli.Command {
...
+	return cli.Command{
+		Name: "download",
+		Flags: append(clientFlags(),
+			cli.StringFlag{
+				Name:  joinFlagNames(urlFlagName, "p"),
+				Usage: "specify the path (URL) of the file to download on the remote.",
+			},
+			cli.StringFlag{
+				Name:  extractPathFlagName,
+				Usage: "if specified, attempt to extract the downloaded artifact",
```
Comment (Kimchelly) [inline cli/command.go]: Can we append " to the given path"?

File: cli/command.go [hunk 5]
```diff
@@ -368,3 +368,64 @@ func KillAllCMD() cli.Command {
 		},
 	}
 }
+
+// DownloadCMD exposes a simple interface for using jasper to download
+// files on the remote jasper.Manager.
+func DownloadCMD() cli.Command {
```
Comment (bsamek) [inline cli/command.go]: This should be `DownloadCmd`.

Comment (tychoish) [reply cli/command.go]: all the other user commands have CMD.

**Pull request merged**